modules/analysis-common/src/main/java/org/opensearch/analysis/common/AbstractCompoundWordTokenFilterFactory.java

```diff
@@ -50,6 +50,8 @@ public abstract class AbstractCompoundWordTokenFilterFactory extends AbstractTok
     protected final int minSubwordSize;
     protected final int maxSubwordSize;
     protected final boolean onlyLongestMatch;
+    protected final boolean noSubMatches;
+    protected final boolean noOverlappingMatches;
```

Comment:
Can you push these fields down into `HyphenationCompoundWordTokenFilterFactory`?

As written, this implicitly adds the settings to `DictionaryCompoundWordTokenFilterFactory` too.

Reply:
Good catch. Will do!